### ISSUE TITLE
script-src CSP was being injected all the time

### ISSUE THREAD

**Issue by robinwhittleton** [rNeomy/javascript-toggle-on-and-off#8]
I just had an issue where the `script-src` CSP was turning up in a site I was developing. Toggling the JS button on and off didn’t do anything, and neither did restarting Firefox (today’s Nightly). Ended up disabling the extension completely and reenabling it, which fixed the problem. Unfortunately I have no idea how it got into that state and can’t replicate it, but worth reporting anyway. Feel free to close if you want.

**Comment by robinwhittleton** [comment]
Happened one more time. Again, no idea how it ended up in the state (although I had been actively using the button).

**Comment by rNeomy** [comment]
So it added the `script-src 'none'` even when the extension was disabled?

**Comment by robinwhittleton** [comment]
Nope, when the extension was disabled it fixed itself, it had just got itself into a state where it was permanently injecting the `script-src` regardless of the state of the button. I had a look through the code and I can’t see any obvious reason why it would get into that state, and it’s possible that it’s a recent bug with Nightly instead of your code. Just worth having a bug open anyway in case other people are seeing the same thing.

**Comment by rNeomy** [comment]
Please update if there is a way to reliably regenerate this. I am keeping this open for others to participate.

**Comment by deant** [comment]
Something similar happened to me:  JS stopped working altogether, regardless of button state.  I had to restart FF to restore normal operation.  Don't know how or why it happened, though!

**Comment by rNeomy** [comment]
The next time it happens, open the browser console (Ctrl + Shift + J) and change the button's state to see if you get any new error when toggling the state.
